src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/markup/RawHtml.java

```diff
@@ -151,14 +151,6 @@ protected static int charCount(CharSequence htmlText) {
                             state = State.ENTITY;
                             count++;
                             break;
-                        case '\r':
-                            // Windows uses "\r\n" as line separator while UNIX uses "\n".
-                            // Skip the "\r" to get consistent results across platforms.
-                            if (i + 1 < htmlText.length() && htmlText.charAt(i + 1) == '\n') {
-                                i++;
-                            }
-                            count++;
-                            break;
```

Comment:
The RawHtml constructor contains `assert Text.checkNewlines(rawHtml)` so I'm not sure this is actually needed.

Reply:
I'm not sure what you are referring to here. Yes, there is the assertion about newlines, but you still have to scan the `rawHtml` for start and end elements, and entities.  Note there is no longer any special handling of newline characters (`\r`, `\n` etc)

Reply:
Sorry, I somehow misread the diff.